shardingsphere-db-protocol/shardingsphere-db-protocol-postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLColumnDescription.java

```diff
@@ -40,10 +47,22 @@
     
     private final int dataFormat = 0;
     
-    public PostgreSQLColumnDescription(final String columnName, final int columnIndex, final int columnType, final int columnLength) {
+    public PostgreSQLColumnDescription(final String columnName, final int columnIndex, final int columnType, final int columnLength, final ResultSetMetaData resultSetMetaData) {
         this.columnName = columnName;
         this.columnIndex = columnIndex;
-        this.typeOID = PostgreSQLColumnType.valueOfJDBCType(columnType).getValue();
+        if (Types.ARRAY == columnType && resultSetMetaData != null) {
+            String columnTypeName = null;
+            try {
+                columnTypeName = resultSetMetaData.getColumnTypeName(columnIndex);
+            } catch (SQLException e) {
+                if (ThreadLocalRandom.current().nextInt(10) == 6) {
```

Comment:
What is this for?

Reply:
Reduce log if exception thrown always. It will be removed, it should not occur.